core/string/ustring.cpp

```diff
@@ -3956,15 +3958,18 @@ String String::format(const Variant &values, const String &placeholder) const {
 				} else {
 					ERR_PRINT(String("STRING.format Inner Array size != 2 ").ascii().get_data());
 				}
-			} else { //Array structure ["RobotGuy","Logis","rookie"]
-				Variant v_val = values_arr[i];
-				String val = v_val;
+				continue;
+			}
+#endif // DISABLE_DEPRECATED
 
-				if (placeholder.contains_char('_')) {
-					new_string = new_string.replace(placeholder.replace("_", i_as_str), val);
-				} else {
-					new_string = new_string.replace_first(placeholder, val);
-				}
+			// Array structure ["RobotGuy", "Logis", "rookie"].
+			Variant v_val = values_arr[i];
+			String val = v_val;
```

Comment:
While you're at it, why not remove the unnecessary copy?

```suggestion
			String val = values_arr[i];
```

Reply:
I'll see what I can do if I can rebase this, because Git doesn't quite catch that most of what has been done in the PR is indent code